packages/syft/src/syft/service/user/user_service.py

```diff
@@ -84,6 +91,207 @@ def create(
+    ) -> SyftSuccess | SyftError:
+        success_msg = (
+            "If the email is valid, we sent a password "
+            + "reset token to your email or a password request to the admin."
+        )
+        result = self.stash.get_by_email(credentials=context.credentials, email=email)
+        # Isn't a valid email
+        if result.is_err():
+            return SyftSuccess(message=success_msg)
+        user = result.ok()
+
+        user_role = self.get_role_for_credentials(user.verify_key)
+        if user_role == ServiceRole.ADMIN:
+            return SyftError(
+                message="You can't request password reset for an Admin user."
+            )
+
+        # Email is valid
+        # Notifications disabled
+        # We should just sent a notification to the admin/user about password reset
+        # Notifications Enabled
+        # Instead of changing the password here, we would change it in email template generation.
+        root_key = self.admin_verify_key()
+        root_context = AuthedServiceContext(server=context.server, credentials=root_key)
+        link = LinkedObject.with_context(user, context=root_context)
+        notifier_service = context.server.get_service("notifierservice")
+        # Notifier is active
+        notifier = notifier_service.settings(context=root_context)
+        notification_is_enabled = notifier.active
+        # Email is enabled
+        email_is_enabled = notifier.email_enabled
+        # User Preferences allow email notification
+        user_allow_email_notifications = user.notifications_enabled[NOTIFIERS.EMAIL]
+
+        # This checks if the user will safely receive the email reset.
+        not_receive_emails = (
+            not notification_is_enabled
+            or not email_is_enabled
+            or not user_allow_email_notifications
+        )
+
+        # If notifier service is not enabled.
+        if not_receive_emails:
+            message = CreateNotification(
+                subject="You requested password reset.",
+                from_user_verify_key=root_key,
+                to_user_verify_key=user.verify_key,
+                linked_obj=link,
+            )
+
+            method = context.server.get_service_method(NotificationService.send)
+            result = method(context=root_context, notification=message)
+
+            message = CreateNotification(
+                subject="User requested password reset.",
+                from_user_verify_key=user.verify_key,
+                to_user_verify_key=root_key,
+                linked_obj=link,
+            )
+
+            result = method(context=root_context, notification=message)
+            if isinstance(result, SyftError):
+                return result
+
+            return SyftSuccess(message=success_msg)
+
+        # Email notification is Enabled
+        # Therefore, we can directly send a message to the user with its new password.
+        message = CreateNotification(
```

Comment:
Should this be inside an else clause of `if not_receive_emails:`

Reply:
Fixed by: [4322021](https://github.com/OpenMined/PySyft/pull/9052/commits/43220215f24f0fc8b1b174ef28d5a26c3973dfda)